internal/cli/atlas_clusters_update.go

```diff
@@ -98,6 +98,25 @@ func (opts *atlasClustersUpdateOpts) patchOpts(out *atlas.Cluster) {
 	if opts.tier != "" {
 		out.ProviderSettings.InstanceSizeName = opts.tier
 	}
+
+	opts.updateLabels(out)
+}
+
+func (opts *atlasClustersUpdateOpts) updateLabels(out *atlas.Cluster) {
+	found := false
+	for _, v := range out.Labels {
+		if v.Key == config.LabelKey && v.Value == config.LabelValue {
+			found = true
+			break
+		}
+	}
+
+	if !found {
```

Comment:
@gssbzn I know you said always but I do not see the use to put the same label if it is already there. Let me know if I have to change this.

Reply:
this is correct, that's why I mentioned as a gotcha in my original comment
One small nit is that we should follow this same approach for create as well, someone could be creating a cluster with a file and defining their own labels so we need to do this kind of check/append for both update and create